aria/orchestrator/context/common.py

```diff
@@ -115,14 +115,19 @@ def __repr__(self):
 
     @contextmanager
     def logging_handlers(self, handlers=None):
+        original_handlers = self.logger.handlers
         handlers = handlers or []
         try:
             for handler in handlers:
                 self.logger.addHandler(handler)
+            for handler in original_handlers:
+                self.logger.removeHandler(handler)
             yield self.logger
         finally:
             for handler in handlers:
```

Comment:
To prevent the original handlers from being duplicated, you should first (in the `finally` block) empty the logger from handlers, and then add the original handlers.